src/lib/dl/dl.c

```diff
@@ -256,6 +256,39 @@ bool PUBLIC zck_copy_chunks(zckCtx *src, zckCtx *tgt) {
     return true;
 }
 
+bool PUBLIC zck_assembly_ctx(zckCtx *src, zckCtx *tgt) {
```

Comment:
I think this would be better named `zck_find_matching_chunks`.  Ideally, it wouldn't bother checking for a match where tgt_idx->valid already == 1.  This way, you could use it to match against one source, then another, and then another.

Reply:
Very good idea - I change exactly as you propose, this will let to use multiple sources.